src/lib.rs

```diff
@@ -25,7 +25,11 @@ pub fn good_fences(opts: GoodFencesOptions) -> Vec<GoodFencesError> {
     println!("beginning file walks");
 
     let dirs_to_walk = &opts.paths.iter().map(|x| x.as_str()).collect();
-    let good_fences_runner = good_fences_runner::GoodFencesRunner::new(tsconfig, dirs_to_walk);
+    let good_fences_runner = good_fences_runner::GoodFencesRunner::new(
+        tsconfig,
+        dirs_to_walk,
+        opts.ignore_external_fences,
```

Comment:
Prefer a 2 member enum to a bool flag here please. I'm not a fan of boolean function arguments in most scenarios for the reasons outlined here: https://ardalis.com/are-boolean-flags-on-methods-a-code-smell. 

The other option is passing a callback but that might tank perf. Callbacks are sometimes inclined by rustc, and sometimes not, and I'm not so sure about the exact scenarios.